src/as/objects/Mesh.ts

```diff
@@ -9,39 +9,58 @@
 
 import {Geometry} from '../core/Geometry'
 import {Material} from '../materials/Material'
-import {Raycaster} from '../core/Raycaster'
+// import {Raycaster, Intersection} from '../core/Raycaster'
 import {Object3D} from '../core/Object3D'
 import {BufferGeometry} from '../core/BufferGeometry'
-import {Intersection} from '../core/Raycaster'
 import {TrianglesDrawModes} from '../constants'
-import {EventDispatcher} from '../core/EventDispatcher'
 import {Vector3} from '../math/Vector3'
 import {Vector2} from '../math/Vector2'
 // import {Sphere} from '../math/Sphere.js'
 // import {Ray} from '../math/Ray.js'
 import {Matrix4} from '../math/Matrix4'
-import {Triangle} from '../math/Triangle'
+// import {Triangle} from '../math/Triangle'
 import {Face3} from '../core/Face3'
 import {Side, TrianglesDrawMode} from '../constants'
-import {MeshBasicMaterial} from '../materials/MeshBasicMaterial'
+import {MeshPhongMaterial} from '../materials/MeshPhongMaterial'
+import {Color} from '../math/Color'
+
+// TODO `<T>` defaults to BufferGeometry for now, but Geometry is easier to use
+// for beginners. We can change the default to Geometry once we implement
+// Geometry, but for now the initial demo will use BufferGeometry.
```

Comment:
what are the blockers for implementing Geo?

Reply:
It's just more work. `Geometry` requires `BufferGeometry`, but the initial demo will work fine with `BufferGeometry`.

Basically, in simple cases, Geometry classes are swappable with BufferGeometry classes. For example, here's the original demo we're aiming for:

https://codepen.io/trusktr/pen/EzBKYM

And here's a fork that is almost identical, but using `BoxBufferGeometry` instead of `BoxGeometry`:

https://codepen.io/trusktr/pen/QWKdJMd

The only difference is on line 26.

---

If you feel adventurous, give it a go if you want! That's actually a good data-oriented class to get experience with that doesn't require crossing the Wasm-JS boundary.